packages/devtools_app/lib/src/screens/network/offline_network_data.dart

```diff
@@ -0,0 +1,69 @@
+// Copyright 2024 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:devtools_shared/devtools_shared.dart';
+
+import '../../shared/http/http_request_data.dart';
+import '../../shared/primitives/utils.dart';
+import 'network_model.dart';
+
+/// Class to encapsulate offline data for the [NetworkController].
+/// It is responsible for serializing and deserializing offline network data.
+class OfflineNetworkData with Serializable {
+  OfflineNetworkData({
+    required this.requests,
+    this.selectedRequestId,
+    this.recording = false,
+  });
+
+  /// Creates an instance of [OfflineNetworkData] from a JSON map.
+  factory OfflineNetworkData.fromJson(Map<String, dynamic> json) {
+    final List<dynamic> requestsJson = json['requests'] ?? [];
+    final requests = requestsJson
```

Comment:
we need to track the socket statistics as well.

Reply:
Yes sure, will add it, I knew it is required but thought of first doing it for http requests to start with, and later add the implementation for socket reqs.